src/SFA.DAS.ApplyService.Web/Controllers/Roatp/RoatpApplicationController.cs

```diff
@@ -890,18 +899,47 @@ private List<Answer> GetAnswersFromForm(Page page)
 
             #region FurtherQuestion_Processing
             // Get all questions that have FurtherQuestions in a ComplexRadio
-            var questionsWithFutherQuestions = page.Questions.Where(x => x.Input.Type == "ComplexRadio" && x.Input.Options != null && x.Input.Options.Any(o => o.FurtherQuestions.Any()));
+            var questionsWithFutherQuestions = page.Questions.Where(x => (x.Input.Type == "ComplexRadio" || x.Input.Type == "ComplexCheckBoxList") && x.Input.Options != null && x.Input.Options.Any(o => o.FurtherQuestions != null && o.FurtherQuestions.Any()));
 
+       
             foreach (var question in questionsWithFutherQuestions)
             {
-                var answerForQuestion = answers.FirstOrDefault(a => a.QuestionId == question.QuestionId);
+                if (question.Input.Type == "ComplexRadio")
+                {
+                    var answerForQuestion = answers.FirstOrDefault(a => a.QuestionId == question.QuestionId);
+
+
+                    // Remove FurtherQuestion answers to all other Options as they were not selected and thus should not be stored
+                    foreach (var furtherQuestion in question.Input.Options
+                        .Where(opt => opt.Value != answerForQuestion?.Value && opt.FurtherQuestions != null)
+                        .SelectMany(opt => opt.FurtherQuestions))
+                    {
+                        foreach (var answer in answers.Where(a => a.QuestionId == furtherQuestion.QuestionId))
+                        {
+                            answer.Value = string.Empty;
+                        }
+                    }
+                }
 
-                // Remove FurtherQuestion answers to all other Options as they were not selected and thus should not be stored
-                foreach (var furtherQuestion in question.Input.Options.Where(opt => opt.Value != answerForQuestion?.Value && opt.FurtherQuestions != null).SelectMany(opt => opt.FurtherQuestions))
+                if (question.Input.Type == "ComplexCheckBoxList")
```

Comment:
Maybe re-write as an else if or case statement, as a question will only have one input type?

Reply:
This could be written into a more simple function as it's only this bit that is different. Would save a lot of copy & paste noise:

`foreach (var answer in answers.Where(a => a.QuestionId == furtherQuestion.QuestionId))
                        {
                            answer.Value = string.Empty;
                        }`

Reply:
On further inspection it's not quite the same logic.